app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt

```diff
@@ -3016,13 +3013,24 @@ class BrowserTabViewModelTest {
     }
 
     private inline fun <reified T : Command> assertCommandNotIssued() {
-        val issuedCommand = commandCaptor.allValues.find { it is T }
-        assertNull(issuedCommand)
+        val defaultMockingDetails = DefaultMockingDetails(mockCommandObserver)
+        if (defaultMockingDetails.invocations.isNotEmpty()) {
+            verify(mockCommandObserver, atLeastOnce()).onChanged(commandCaptor.capture())
+            val issuedCommand = commandCaptor.allValues.find { it is T }
+            assertNull(issuedCommand)
+        } else {
+            assertTrue(true)
+        }
     }
 
     private inline fun <reified T : Command> assertCommandIssuedTimes(times: Int) {
-        val timesIssued = commandCaptor.allValues.count { it is T }
-        assertEquals(times, timesIssued)
+        if (times == 0) {
+            assertCommandNotIssued<T>()
+        } else {
+            verify(mockCommandObserver, atLeastOnce()).onChanged(commandCaptor.capture())
```

Comment:
doesn't `verify` already provide support for what you're doing here?

`verify(mockCommandObserver, times(times)).onChanged(commandCaptor.capture())` ?

Reply:
Not really. Here we are asserting that a specific command of type `T` has not been emitted. 

`verify(mockCommandObserver, times(times)).onChanged(commandCaptor.capture())` will check any command emitted.

Reply:
`it is T`

ah yeah, missed that detail